File: fluentcore/src/main/java/com/mcreater/fxfluent/xaml/XAMLLoader.java
package com.mcreater.fxfluent.xaml;

import com.google.gson.Gson;

import java.io.IOException;
import java.io.InputStreamReader;
import java.util.List;
import java.util.Map;

public class XAMLLoader {
    public static void loadAll() throws IOException {
        Map map = new Gson().fromJson(new InputStreamReader(XAMLLoader.class.getClassLoader().getResourceAsStream("xaml/indexes/resources.json")), Map.class);
        ((List<String>) map.get("paths")).forEach(XAMLManager::addFileFromClasspath);
        XAMLManager.parse();
    }

    public static void load(String path) {
        XAMLManager.addFileFromClasspath(path);
        XAMLManager.parse();
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/xaml/XamlLoader.kt
package com.mcreater.fxfluent.xaml

import com.google.gson.Gson
import java.io.IOException
import java.io.InputStreamReader

class XamlLoader {
    companion object {
        @JvmStatic
        @Throws(IOException::class)
        fun loadAll() {
            val map = Gson().fromJson<Map<*, *>>(
                XamlLoader::class.java.getClassLoader().getResourceAsStream("xaml/indexes/resources.json")?.let {
                    InputStreamReader(it)
                }, MutableMap::class.java
            )
            (map["paths"] as List<*>?)!!.forEach {
                XamlManager.addFileFromClasspath(it as String)
            }
            XamlManager.parse()
        }
        @JvmStatic
        fun load(path: String?) {
            XamlManager.addFileFromClasspath(path)
            XamlManager.parse()
        }
    }
}